src/routing/core.rs

```diff
@@ -2204,7 +2238,7 @@ impl Core {
         // We need to construct a proof that would be trusted by the destination section.
         let known_key = self
```

Comment:
this shall be `our` key that trusted by destination section.
not `destination section's key` .